### ISSUE TITLE
Implement /test to do dry-run or send to alternate email address

### ISSUE THREAD

**Issue by nasamuffin** [gitgitgadget/gitgitgadget#83]
As discussed in IRC today:

```
17:08:04    nasamuffin_ | I'm trying to figure out whether it makes sense to target the tutorial to using dscho's gitgitgadget & was curious whether there's a way to have it send 
                        | the mails to alternate email                                                                                                                             
17:08:41       jrnieder | nasamuffin_: interesting. Even outside the tutorial context, sounds useful as a sort of "dry run"                                                        
17:08:50        __rohit | nasamuffin_: Yes, a tutorial on gitgitgadget will benefit the beginners                                                                                  
17:09:22          dscho | jrnieder exciting news about trace2! Do you think you will be able to share some of your findings in the future?                                         
17:09:40          dscho | Welcome, nasamuffin_!                                                                                                                                    
17:09:44        __rohit | although current "documentation" of gitgit is good enough                                                                                                
17:10:07          dscho | Regarding alternate email: GitGitGadget is all open source ;-)                                                                                           
17:10:13       jrnieder | dscho: yep! I was happy to see the stats jobriggs described in Git Merge, would love to see more people report that kind of thing more regularly         
17:10:16    nasamuffin_ | it seems straightforward to use, but i don't want newbies spamming the list with a toy patch                                                             
17:10:18         rafasc | On the topic of helping beginners, I have recently discovered https://git-send-email.io/. (I am not related to the project in any shape or form)         
17:10:25    nasamuffin_ | dscho: i'll learn typescript and figure it out :P                                                                                                        
17:10:49          dscho | __rohit: please do open PRs to improve the documentation as you see fit.                                                                                 
17:11:11          dscho | nasamuffin_ I often hang out in this here channel, and try to do my best to answer questions ;-)                                                         
17:11:20  jonathantanmy | For me, more protocol v2 bug fixes and a small partial clone improvement (batch blobs fetched for diff/show). Also worked on some CDN offloading patches 
                        | for JGit, but I don't know when that will see production.                                                                                                
17:12:20          dscho | rafasc is that site really just accumulating links to real documentation?                                                                                
17:12:41        __rohit | dscho: I will.                                                                                                                                           
17:12:49       jrnieder | jonathantanmy: you're burying the lede. ;-) Interested reviewers can look at the series ending at https://git.eclipse.org/r/c/139993                     
17:12:50          dscho | nasamuffin_ right, I think a `/test` command would be useful, which should send the series to yourself only.                                             
17:13:30          dscho | nasamuffin_ for that, we would need the information for each allowed user what their desired email address is (we should have that information anyway,   
17:13:30          dscho | nasamuffin_ for that, we would need the information for each allowed user what their desired email address is (we should have that information anyway,   
                        | to Cc: the cover letter to that address).                                                                                                                
17:13:52          dscho | nasamuffin_ so it looks like a slightly bigger than micro-project, but totally doable (and killing two birds with the same stone).                       
```

There seems to be a use case for a /test or similar which performs a dry-run of the mail patches to a submitter-provided email for inspection before heading to the broader list..

**Comment by webstech** [comment]
How about /preview for the command name? /test implies something else to me.  

Possible implementation:
1. ci-helper::handleComment would pass the email to gitgitGadget::submit
2. submit would add it to the smtpOptions if passed
3. send-mail::sendMail would override the 'to' and not add the 'cc' if the smtpOptions had the destination override.

I don't think using ISMTPOptions is really a hack since it does have mailing related information.  Another approach would be to just pass another parm on the call to send-mail::parseHeadersAndSendMail and sendMail.  Not a big fan of this method since more functions have to be updated for an infrequently (?) used feature.

Open to other suggestions of course.

**Comment by dscho** [comment]
> How about /preview for the command name? /test implies something else to me.

Or even `/send <email>`? Of course, we would only accept this command from a registered user.

**Comment by dscho** [comment]
> Possible implementation:
> 
>     1. ci-helper::handleComment would pass the email to gitgitGadget::submit
> 
>     2. submit would add it to the smtpOptions if passed
> 
>     3. send-mail::sendMail would override the 'to' and not add the 'cc' if the smtpOptions had the destination override.
>
> [...]
> 
> Forgot about patch-series::generateAndSend updating notes, etc. A new option similar to dryRun could be used.

That sounds very good!

I think we might also need to disable the check whether the current commits have been submitted already...

**Comment by webstech** [comment]
PR #125 has been opened.  I chose to stay with /preview as the comment/command.  I did not implement the option to provide an email address (while it is a controlled group, it still could allow for sending unwanted emails). The requester can forward the emails.  The check on already submitted was put in.

Added a flag to indicate no update where dryRun was previously used.  dryRun is still used for some things.  The user of dryRun now also sets noUpdate.

Tested using [ethereal.email](https://ethereal.email).  That requires some config support that I will submit in a separate PR - not ethereal specific but to allow for alternate mail servers for testing.
